rdrf/rdrf/models/proms/models.py

```diff
@@ -153,6 +161,59 @@ def expression(self):
         else:
             return self.cde.name + "  if " + self.precondition.cde.name + " = " + self.precondition.value
 
+    def clean(self):
+
+        if self.cde.code != "PROMSConsent":
+
+            if self.cde_path:
+                # Extract form and section code from /FROM_NAME/SECTION_CODE/.
+                path_values = list(filter(None, self.cde_path.split("/")))
+
+                # Check the path contain a form_name and section_code, and only these exact two variables.
+                if len(path_values) != 2:
+                    raise ValidationError(f"[{self.cde.code}] The path '{self.cde_path}' is not properly formatted - it should contains exactly one form name and one section code separated by slashes: \"/FORM_NAME/SECTION_CODE/\"")
+                path_form_name, path_section_code = path_values
+
+                # Check that the form_name exist for the selected registry.
+                try:
+                    path_form = RegistryForm.objects.get(name=path_form_name, registry=self.survey.registry)
+                except (RegistryForm.DoesNotExist, RegistryForm.MultipleObjectsReturned):
+                    raise ValidationError(f"[{self.cde.code}] The form '{path_form_name}' doesn't exist the selected registry {self.survey.registry.code}")
+
+                # Check that the section name exist for this form_name.
+                if path_section_code not in path_form.sections.split(","):
+                    raise ValidationError(f"[{self.cde.code}] The section '{path_section_code}' does not exist in the form '{path_form_name}'")
+
+                # Check that the cde exist for this section.
+                if self.cde.code not in Section.objects.get(code=path_section_code).get_elements():
+                    raise ValidationError(
+                        f"[{self.cde.code}] The cde {self.cde.code} does not exist in the form '{path_form_name}' / section '{path_section_code}'")
+            else:
+                # Check that a default form is selected for this survey.
```

Comment:
refactor a bit to compress this code - helper functions or methods  ( _helper_method   )